artcommon/artcommonlib/util.py

```diff
@@ -430,3 +430,15 @@ def detect_package_managers(metadata, dest_dir: Path):
         if any(dest_dir.joinpath(file).is_file() for file in files):
             pkg_managers.append(pkg_manager)
     return pkg_managers
+
+
+@retry(reraise=True, wait=wait_fixed(10), stop=stop_after_attempt(3))
+async def get_konflux_slsa_attestation(pull_spec: str):
+    """
+    Inspect the SLSA provenance: https://konflux.pages.redhat.com/docs/users/metadata/attestations.html
+    To see if the latest build is actually hermetic
+    """
+    cmd = f"cosign download attestation {pull_spec} --registry-username {os.environ['KONFLUX_ART_IMAGES_USERNAME']} --registry-password {os.environ['KONFLUX_ART_IMAGES_PASSWORD']}"
```

Comment:
would recommend readin env vars at the beginning of cli script and then passing them along as normal func params to this func.

Reply:
Not sure where you mean beginning at the cli script? Do you mean as part of the `click` params?

Reply:
For example - https://github.com/openshift-eng/art-tools/blob/543b5a99dbb0e8daf2bc9cf92c45988cf5233735/elliott/elliottlib/cli/konflux_release_cli.py#L259

Reply:
thanks - updated in https://github.com/openshift-eng/art-tools/pull/1570/commits/83dc5bb61da5145df894dfc287fca0dbe191aef9

so the goal is try to get the variable in the beginning and fail fast if it doesn't